code/modules/clothing/clothing_accessories.dm

```diff
@@ -62,9 +62,13 @@
 	if(!(A in accessories))
 		return
 
-	A.on_removed(user)
-	accessories -= A
-	update_wear_icon()
+	if(istype(A,/obj/item/clothing/accessory/armor))
+		to_chat(user, SPAN_WARNING("Removing this accessory would ruin it."))
```

Comment:
Would let any accessory have a variable that doesn't let it be removed instead of it being hard coded to the armor path.